deploy/helm/sumologic/upgrade-1.0.0.sh

```diff
@@ -312,6 +312,70 @@ for i in $(seq 0 ${metrics_length}); do
     fi
 done
 
+# Fix fluent-bit env
+fb_length="$(yq r -l "new.yaml" 'fluent-bit.env')"
+fb_length="$(( ${fb_length} - 1))"
+
+for i in $(seq 0 ${fb_length}); do
+  env_name="$(yq r "new.yaml" "fluent-bit.env[${i}].name")"
+  if [[ "${env_name}" = "CHART" ]]; then
+    yq w -i new.yaml "fluent-bit.env[${i}].valueFrom.configMapKeyRef.key" "fluentdLogs"
+  fi
+done
+
+# Fix prometheus service monitors
+yq d -i "new.yaml" "prometheus-operator.prometheus.additionalServiceMonitors(name==collection-sumologic)"
+yq d -i "new.yaml" "prometheus-operator.prometheus.additionalServiceMonitors(name==collection-sumologic-otelcol)"
+echo '---
+prometheus-operator:
+  prometheus:
+    additionalServiceMonitors:
+      - name: collection-sumologic-otelcol
+        additionalLabels:
+          app: collection-sumologic-otelcol
+        endpoints:
+          - port: metrics
+        namespaceSelector:
+          matchNames:
+            - sumologic
+        selector:
+          matchLabels:
+            app: collection-sumologic-otelcol
+      - name: collection-sumologic-fluentd-logs
+        additionalLabels:
+          app: collection-sumologic-fluentd-logs
+        endpoints:
+        - port: metrics
+        namespaceSelector:
+          matchNames:
+          - sumologic
+        selector:
+          matchLabels:
+            app: collection-sumologic-fluentd-logs
+      - name: collection-sumologic-fluentd-metrics
+        additionalLabels:
+          app: collection-sumologic-fluentd-metrics
+        endpoints:
+        - port: metrics
+        namespaceSelector:
+          matchNames:
+          - sumologic
+        selector:
+          matchLabels:
+            app: collection-sumologic-fluentd-metrics' | yq m -a -i "new.yaml" -
+service_mon_length="$(yq r -l "new.yaml" 'prometheus-operator.prometheus.additionalServiceMonitors')"
+service_mon_length="$(( ${service_mon_length} - 1))"
+for i in $(seq 0 ${service_mon_length}); do
+  service_mon_name="$(yq r "new.yaml" "prometheus-operator.prometheus.additionalServiceMonitors[${i}].name")"
+  if [[ "${service_mon_name}" = "collection-sumologic-events" ]]; then
```

Comment:
I don't understand why we have a special case for events servicemonitor, is it not possible to handle with the delete or merge you did above?

Reply:
I started from that approach, and later I get into two exceptions:
 * splitting collection-sumologic on logs and metrics
 * strange behavior of otelcol which was duplicated after merge